### PR TITLE
Drag changes

### PR DESCRIPTION
**Pull request by metalgearsloth**
* Higlights only show near cursor which seems nicer than highlighting the whole screen.
* Don't highlight un-droppable entities.
* Disables interaction outlines while dragging so they're not overlaying over each other.
* Fixes https://github.com/space-wizards/space-station-14/issues/2311

I moved a bunch of stuff to shared so the client has a better if when a drag's valid. It won't be 100% perfect as not all of it's predicted but it should fix the range issues.

### CODE REVIEW

**Comment by metalgearsloth** [comment]
Should be good for a review again

**Comment by metalgearsloth** [comment]
Merging in a week if no objections

File: Content.Server/GameObjects/Components/Disposal/DisposalUnitComponent.cs [hunk 1]
```diff
@@ -139,15 +141,14 @@ private bool Engaged
 
         public bool CanInsert(IEntity entity)
```
Comment (PaulRitter) [inline Content.Server/GameObjects/Components/Disposal/DisposalUnitComponent.cs:142]:
```suggestion
        public override bool CanInsert(IEntity entity)
```

File: Content.Server/GameObjects/Components/Disposal/DisposalUnitComponent.cs [hunk 2]
```diff
@@ -6,13 +6,15 @@
...
 using Content.Server.GameObjects.Components.GUI;
+using Content.Server.GameObjects.Components.Power.ApcNetComponents;
 using Content.Server.GameObjects.Components.Items.Storage;
 using Content.Server.GameObjects.Components.Power.ApcNetComponents;
 using Content.Server.GameObjects.EntitySystems;
 using Content.Server.GameObjects.EntitySystems.DoAfter;
 using Content.Server.Interfaces;
 using Content.Server.Interfaces.GameObjects.Components.Items;
 using Content.Server.Utility;
+using Content.Shared.GameObjects.Components.Disposal;
```
Comment (PaulRitter) [inline Content.Server/GameObjects/Components/Disposal/DisposalUnitComponent.cs:17]:
```suggestion
using Content.Shared.GameObjects.Components.Disposal;
```
duplicate

**Review by PaulRitter — CHANGES_REQUESTED** [review]
only looked at this in the web reviewer, didn't checkout code locally

**Pull request merged**